eox_core/api/v1/views.py

```diff
@@ -665,6 +669,95 @@ def log(self, desc, data, exception=None):
         LOG.error(' '.join(log_data))
 
 
+class EdxappGrade(UserQueryMixin, APIView):
+    """
+    Handles API requests to manage course grades
+    """
+    authentication_classes = (BearerAuthentication, SessionAuthentication)
+    permission_classes = (EoxCoreAPIPermission,)
+    renderer_classes = (JSONRenderer, BrowsableAPIRenderer)
+
+    def get(self, request):
+        """
+        Retrieves Grades information for given a user and course_id
+
+        **Example Requests**
+
+            GET /eox-core/api/v1/grade/?username=johndoe&
+            course_id=course-v1:edX+DemoX+Demo_Course
+
+            Request data: {
+              "username": "johndoe",
+              "course_id": "course-v1:edX+DemoX+Demo_Course",
+            }
+
+        **Returns**
+
+        - 200: Success.
+        - 400: Bad request, missing course_id or either email or username.
+        - 404: User, course or enrollment not found.
+        """
+        user_query = self.get_user_query(request)
+        user = get_edxapp_user(**user_query)
+
+        course_id = self.query_params.get('course_id', None)
+        detailed = self.query_params.get('detailed', False)
+        grading_policy = self.query_params.get('grading_policy', False)
+
+        if not course_id:
+            raise ValidationError(detail='You have to provide a course_id')
+
+        _, errors = get_enrollment(username=user.username, course_id=course_id)
+
+        if errors:
+            raise NotFound(errors)
+
+        grade_factory = get_course_grade_factory()
+        course_key = get_valid_course_key(course_id)
+        course = get_courseware_courses().get_course_by_id(course_key)
+        course_grade = grade_factory().read(user, course)
+        response = {"earned_grade": course_grade.percent}
+
+        if detailed:
+            breakdown = self._section_breakdown(course_grade.subsection_grades)
+            response['section_brekdown'] = breakdown
```

Comment:
can you use just double quotes or single, not both

Reply:
I suggest you to run [black](https://github.com/psf/black) on this file

Reply:
Done. But I let the imports as they were before to avoid isort warnings. If we end up adopting black in the future maybe we should look into our style configs?